lib/handlebars/compiler/compiler.js

```diff
@@ -323,18 +323,28 @@ Compiler.prototype = {
 
   Hash: function(hash) {
     let pairs = hash.pairs,
-        i = 0,
+        i = 0, splat = null,
         l = pairs.length;
 
     this.opcode('pushHash');
-
     for (; i < l; i++) {
+
+      if (pairs[i].type === 'Splat') {
+        if (splat !== null) {
+          throw new Exception('Multiple splats are not supported yet');
```

Comment:
I think this is a fundamental issue that needs to be resolved before we can merge it. Support for multiple instances is something that will be confusing if it's not there but values can be assigned on either side.

It seems like we need to add finer tuned opcodes that only generate objects, i.e. something like:

```
pushHash
pushObject
{pushParam}
{pushParam}
assignToHash
assignToHash
popObject
splat
pushObject
{pushParam}
assignToHash
popObject
popHash
```

Which would end up generating javascript similar to the following:

```
hash: this.splat({key: value, key: value}, splat, {key: value})
```